docs/development.md

```diff
@@ -35,3 +35,20 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Cross-Platform Scripting
```

Comment:
All the information here is good, but I think we need to change up the style so it is better for people looking at it quickly as reference material.
They will usually just want to look up things quickly and not read a paragraph.

I would take a look at Google Style guides for reference.
The [C++ guide](https://google.github.io/styleguide/cppguide.html) is the gold standard, 
but I think it is a bit detailed for our case.
The [Golang guide](https://google.github.io/styleguide/go/best-practices) would probably be a closer (but not exact) match.